auth/role_manager.hh

```diff
@@ -159,31 +159,31 @@ public:
     ///   (role2, role3)
     /// }
     ///  
-    virtual future<role_to_directly_granted_map> query_all_directly_granted() = 0;
+    virtual future<role_to_directly_granted_map> query_all_directly_granted() const = 0;
 
-    virtual future<role_set> query_all() = 0;
+    virtual future<role_set> query_all() const = 0;
 
-    virtual future<bool> exists(std::string_view role_name) = 0;
+    virtual future<bool> exists(std::string_view role_name) const = 0;
 
     ///
     /// \returns an exceptional future with \ref nonexistant_role if the role does not exist.
     ///
-    virtual future<bool> is_superuser(std::string_view role_name) = 0;
+    virtual future<bool> is_superuser(std::string_view role_name) const = 0;
 
     ///
     /// \returns an exceptional future with \ref nonexistant_role if the role does not exist.
     ///
-    virtual future<bool> can_login(std::string_view role_name) = 0;
+    virtual future<bool> can_login(std::string_view role_name) const = 0;
 
     ///
     /// \returns the value of the named attribute, if one is set.
     ///
-    virtual future<std::optional<sstring>> get_attribute(std::string_view role_name, std::string_view attribute_name) = 0;
+    virtual future<std::optional<sstring>> get_attribute(std::string_view role_name, std::string_view attribute_name) const = 0;
 
     ///
     /// \returns a mapping of each role's value for the named attribute, if one is set for the role.
     ///
-    virtual future<attribute_vals> query_attribute_for_all(std::string_view attribute_name) = 0;
+    virtual future<attribute_vals> query_attribute_for_all(std::string_view attribute_name) const = 0;
```

Comment:
Unfortunately (and surprisingly) it does, thank you. The commit will need to be dropped

Reply:
in this case since it's an interface it's a bit difficult to predict which implementation is const and which not (and I am not sure how adding it helps, it mostly just accidentally locks the implementation)

Reply:
That was actually part of the rationale for going with those changes. If something turned out to need to use a field of the role manager, it should be marked as mutable. Another reason was to be able to describe auth on a const-qualified service/role_manager; it having to be mutable would be confusing. However, I had my doubts if that was the right decision too, and it seems it was. I'll revert the commits and adjust the rest of the code